### PR TITLE
Cockpit rhel host parametrization

### PR DESCRIPTION
**Pull request by peterdragun**
~~Merge after: SatelliteQE/airgun/pull/676~~ :heavy_check_mark: 

Adding parametrization for cockpit content host testing.

### CODE REVIEW

**Comment by peterdragun** [comment]
trigger: test-robottelo
pytest: tests/foreman/ui/test_host.py -k "test_positive_cockpit[rhel7] or test_positive_cockpit[rhel8]"
airgun: 676

File: pytest_fixtures/core/contenthosts.py [hunk 1]
```diff
@@ -107,21 +107,45 @@ def registered_hosts(organization_ak_setup, content_hosts, default_sat):
...
         organization=module_org, name=f'rhel{rhelver}_{gen_string("alpha")}'
     ).create()
-    sattools_repo = entities.Repository(
-        organization=module_org,
-        product=prod,
-        content_type='yum',
-        url=settings.repos['SATCLIENT_REPO'][f'RHEL{rhelver}'],
-    ).create()
-    sattools_repo.sync()
+    for url in repo_urls:
```
Comment (tstrych) [inline pytest_fixtures/core/contenthosts.py:138]: why do you need  repo_urls here ?
isn't the one repo per version enough ?

In the previous version you enable one repo per version, 
you don't need more of them right ? for one rhel version one custom satclient repo

File: robottelo/hosts.py [hunk 2]
```diff
@@ -413,6 +413,16 @@ def install_capsule_katello_ca(self, capsule=None):
         if result.status != 0:
             raise ContentHostError('Failed to install the katello-ca rpm')
 
+    def install_cockpit(self):
+        """Installs cockpit on the broker virtual machine.
```
Comment (tstrych) [inline robottelo/hosts.py:417]: Install cockpit on the Content Host.

(no need for broker virtual machine)

**Review by tstrych — APPROVED** [review]
ACK, pending comments.

please also provide test Result for RHEL9.

File: pytest_fixtures/core/contenthosts.py (hunk 1 above)
Comment (peterdragun) [reply pytest_fixtures/core/contenthosts.py:138]: The cockpit on RHEL 7 requires additional extras, but those need base os RPMs to install correctly.

File: robottelo/hosts.py (hunk 2 above)
Comment (peterdragun) [reply robottelo/hosts.py:417]: Just wanted to use a similar docstring to other functions in this module

File: pytest_fixtures/core/contenthosts.py [hunk 3]
```diff
@@ -107,21 +107,45 @@ def registered_hosts(organization_ak_setup, content_hosts, default_sat):
...
+@pytest.fixture(scope="function")
+def cockpit_host(default_sat, module_org, rhel_contenthost):
+    """Register content host to Satellite and install cockpit on the host."""
+    rhelver = rhel_contenthost.os_version.major
+    if rhelver > 7:
+        repo = [settings.repos[f'rhel{rhelver}_os']['baseos']]
+    else:
+        repo = [settings.repos['rhel7_os'], settings.repos['rhel7_extras']]
+    register_host_custom_repo(default_sat, module_org, rhel_contenthost, repo)
+    rhel_contenthost.execute(f"hostnamectl set-hostname {rhel_contenthost.hostname} --static")
```
Comment (pondrejk) [inline pytest_fixtures/core/contenthosts.py:125]: just curious, why this is needed? Isn't this pointing to some problem at satlab side?

File: tests/foreman/ui/test_host.py [hunk 4]
```diff
@@ -2318,9 +2318,10 @@ def test_positive_gce_cloudinit_provision_end_to_end(
 
 @pytest.mark.destructive
 @pytest.mark.upgrade
+@pytest.mark.rhel_ver_match('[^6].*')
```
[the remark this answers is not in the document]
Comment (pondrejk) [reply tests/foreman/ui/test_host.py:2321]: In this case I think it makes sense to include fips hosts to such integration tests. If we checked only for the webconsole button then no, but the actual communication between host and the UI could possibly be influenced by stricter fips rules, so I think it's good to have that coverage

**Review by pondrejk — APPROVED** [review]

File: pytest_fixtures/core/sys.py [hunk 5]
```diff
@@ -1,6 +1,7 @@
 from urllib.parse import urlparse
 
 import pytest
+from nailgun import entities
```
Comment (Gauravtalreja1) [inline pytest_fixtures/core/sys.py:4]: Instead of importing nailgun here, you could call like `default_sat.cli.Setting().###`

Comment (mshriver) [reply pytest_fixtures/core/sys.py:4]: @Gauravtalreja1 close! To keep things on the api, use `default_sat.api.Setting` to reference the same class from `nailgun.entities.Setting`.

File: pytest_fixtures/core/contenthosts.py (hunk 3 above)
Comment (peterdragun) [reply pytest_fixtures/core/contenthosts.py:125]: The cockpit is using static hostname in UI, so without this command, it is reporting localhost. Not sure if we want to set the static hostname on all hosts from satlab. Without this command transient hostname is still set to the correct value.

**Pull request merged**